tensorbay/opendataset/AADB/loader.py

```diff
@@ -0,0 +1,106 @@
+}
+
+_IMAGE_ROOT_PATH = {
+    "new_test": "AADB_newtest",
+    "test": "datasetImages_warp256",
+    "train": "datasetImages_warp256",
+    "val": "datasetImages_warp256",
+}
+
+_DATASET_NAME = "AADB"
+
+
+def AADB(path: str) -> Dataset:
+    """Load the AADB to TensorBay.
+
+    The file structure looks like:
+    <path>
+        AADB_newtest/
+            0.500_farm1_487_20167490236_ae920475e2_b.jpg
+            ...
+        datasetImages_warp256/
+            farm1_441_19470426814_baae1eb396_b.jpg
+            ...
+        imgListFiles_label/
+            imgList<segment_name>Regression_<attribute_name>.txt
+            ...
+
+    Arguments:
+        path: the root directory of the dataset
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(_DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    attributes_info = dataset.catalog.classification.attributes
+
+    for mode, image_root_path in _IMAGE_ROOT_PATH.items():
+        image_name_handler = (
+            _new_test_image_name_handler if mode == "new_test" else _image_name_handler
+        )
+
+        segment = dataset.create_segment(mode)
+        image_name_attributes_map = _convert_txt_to_dict(root_path, mode, attributes_info)
+        for image_name, attributes in image_name_attributes_map.items():
+            image_path = os.path.join(
+                root_path, image_root_path, image_name_handler(image_name, attributes)
+            )
+            data = Data(image_path)
+            data.label.classification = Classification(attributes=attributes)
+            segment.append(data)
+
+    return dataset
+
+
+def _image_name_handler(image_name: str, _: Dict[str, float]) -> str:
+    return image_name
+
+
+def _new_test_image_name_handler(image_name: str, attributes: Dict[str, float]) -> str:
+    # the image name of new_test under `AADB_newtest` looks like
+    # 0.500_farm1_487_20167490236_ae920475e2_b.jpg
+    # 0.500 is the score of the image
+    # the image name of new_test in label file looks like
+    # farm1_487_20167490236_ae920475e2_b.jpg
+    return f"{attributes['score']:.3f}_{image_name}"
+
+
+def _convert_txt_to_dict(
+    path: str, mode: str, attributes_info: NameList[AttributeInfo]
+) -> Dict[str, Dict[str, float]]:
+    image_name_attributes_map: DefaultDict[str, Dict[str, float]] = defaultdict(dict)
+    label_root_path = os.path.join(path, "imgListFiles_label")
+    for attribute_name in attributes_info.keys():
+        label_file_name = f"{_LABEL_FILE_PREFIX[mode]}_{attribute_name}.txt"
+        label_file_path = os.path.join(label_root_path, label_file_name)
+        with open(label_file_path) as fp:
+            # one line of file looks like:
+            # "<image_name> score"
+            for line in fp:
+                image_name, score = line.strip().split()
```

Comment:
what is the meaning of the variable name `score`? is the same as the attribute `score`?